react/QuantitySelector.js

```diff
@@ -0,0 +1,40 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+
+/**
+* Quantity selector component.
+*/
+class QuantitySelector extends Component {
+  handleChange = (event) => {
+    const { maxQuantity, onQuantityChange, onMaxReached } = this.props
+    let value = event.target.value
+    let quantity = value ? parseInt(value, 10) : 0
+    if (quantity > maxQuantity) {
+      quantity = maxQuantity
+      onMaxReached()
+    } else if (quantity < 0) {
+      quantity = 0
+    }
+    this.setState({ currentQuantity: quantity })
```

Comment:
Is this `currentQuantity` state variable ever used?
The `render` method uses the one that is inside the `props`.